dhis-2/dhis-services/dhis-service-administration/src/main/resources/data-integrity-checks/tracked_entity_attributes/tracked_entity_attributes_trigram_index_overview.yaml

```diff
@@ -0,0 +1,41 @@
+---
+name: tracked_entity_attributes_trigram_index_overview
+description: Tracked entity attributes that have a partial trigram index created on the trackedentityattributevalue table.
+section: Tracked entity attributes
+section_order: 3
+summary_sql: >-
+    WITH existing_trigram_indexes AS (
+        SELECT
+        CAST(substring(indexdef FROM 'trackedentityattributeid\s*=\s*(\d+)') AS bigint) AS teaid
+        FROM pg_indexes
+        WHERE tablename = 'trackedentityattributevalue'
+        AND indexdef ILIKE '%gin_trgm_ops%'
+        AND indexdef ILIKE '%WHERE%'
+        AND indexdef ~ '[(]?\s*trackedentityattributeid\s*=\s*\d+\s*[)]?'
+    )
+    SELECT COUNT(*) AS value, 100 * COUNT(*) / NULLIF((SELECT COUNT(*) FROM trackedentityattribute), 0) AS percent
+    FROM trackedentityattribute tea
+    WHERE tea.trackedentityattributeid IN (SELECT teaid FROM existing_trigram_indexes);
+details_sql: >-
+    WITH existing_trigram_indexes AS (
+        SELECT
+        CAST(substring(indexdef FROM 'trackedentityattributeid\s*=\s*(\d+)') AS bigint) AS teaid
+        FROM pg_indexes
+        WHERE tablename = 'trackedentityattributevalue'
+        AND indexdef ILIKE '%gin_trgm_ops%'
+        AND indexdef ILIKE '%WHERE%'
+        AND indexdef ~ '[(]?\s*trackedentityattributeid\s*=\s*\d+\s*[)]?'
+    )
+    SELECT uid, name
+    FROM trackedentityattribute tea
+    WHERE tea.trackedentityattributeid IN (SELECT teaid FROM existing_trigram_indexes);
+details_id_type: trackedEntityAttributes
+severity: INFO
+introduction: >
+    A partial trigram index can be created on the `trackedentityattributevalue` table for a Tracked Entity Attribute (TEA) if the TEA has the trigram indexable flag set to true and supports at least one of the operators: `LIKE` or `EW`.
```

Comment:
Should we also mention that minimum char must be greater or equal 3?

Reply:
The min characters >= 3 is only required when using the index. To create it we just consider the flag and the blocked operators.
The previous [integrity check](https://github.com/dhis2/dhis2-core/pull/21631/files#diff-f4997319e8020e9e48c8cdff38fafecbe7e5c70f4d8d87a7c545676a99ac13edR28), where we validate the TEA configuration lets the user know if the min characters are set below 3.